### DISCUSSION TITLE
Qt version

### DISCUSSION THREAD

**Discussion by isf63** [hardinfo2/hardinfo2#86] [General]
I'm wondering why Hardinfo2 depends on Qt5 when it uses GTK for its GUI. I know that it can be compiled without any Qt, but most distros do not provide that option.

Also, any plans on transitioning to Qt6?

**Comment by hwspeedy** [comment]
Hi isf63,

QT5 is needed for qgears2 OpenGL benchmark.

QT6 is too new for older distros and there are other QT5 apps that most people use, so these libs will probably be used anyway. 

Is that why you ask?

**Reply by isf63** [reply]
I see, thanks.

I naively wanted to make progress toward ridding my system of Qt5. The transition from Qt5->Qt6 is going to leave most people with both on their system, so it doesn't really matter.

**Comment by uro666** [comment]
Hello @hwspeedy 

I am a package maintainer for [OpenMandriva Lx](https://www.openmandriva.org/) and saw this discussion about Qt6 & rather than open an issue opted to post here.

We have moved over to Qt6 for quite a long time now and are in the ongoing process of dropping Qt5 as it is out of support and out of maintenance, we have been moving packages over to Qt6 where possible and looking for alternatives where that is not possible.

Today I updated our `hardinfo2` package and as we are dropping Qt5 I had to disable the `qgears2` benchmark in hardinfo2 due to its dependency on Qt5 libraries by using the build option to do that (thank you for providing that), we have the `vkgears` module enabled and building as an alternative.

It would be great if `qgears2` was ported to Qt6 in order that we can re-enable the module and keep our users providing OpenGL gears benchmark data for the project.

As a suggestion:  it should be possible to have two versions of `qgear2` in `hardinfo2`, one Qt5 version and one ported to Qt6 (selectable via CMake build options) and that way you could continue to support users on distros still using Qt5 as well adding support for distros that have moved on to using Qt6.

I appreciate that would mean extra work for you to port it, however it would mean that distros using Qt6 could offer the full range of benchmarks contained in `hardinfo2` and keep your benchmark datasets more complete.

Thank you for your time.

**Reply by hwspeedy** [reply]
@uro666,

Yes, that will probably need some considerations - this is one of the problems with Linux Desktop.

The best solution would be going OpenGL native - but I have no skills in that area. **Any one who can help? - would also remove QT5 120MB framework!**

The QT6 patch was low and can be ifdef'ed to allow for new and old distros.
PS: We cannot have both QT5 and QT6 versions of qgear2 - if that was possible - we would just stay at QT5! - so each distro release is either QT5 or QT6.

But we should probably note the QT5 vs QT6 in the benchmark to avoid cluttering results.

/hwspeedy

**Comment by hwspeedy** [comment]
@uro666

**Hardinfo2 - Regarding QT6 only distro (Mandriva next) - see https://github.com/hardinfo2/hardinfo2/commit/463e5c1fc314d3f43c0593ce5fe881ffe6c04abe**
NOTE: QT is just used for the OpenGL Benchmark - Program uses GTK3.

This will be in next release after some more testing - **Please do not change for Mandriva 6.0, that supports QT5 also**.

When we change to QT6 we start fresh on new distros only - no compare any more for OpenGL benchmark for history and common ground for all Linux lovers. 
**So stay on QT5 when running on dual stack QT5/QT6 distro.**

NOTE: QT6 is very much slower than QT5 - Hence experimental.

When will the next QT6 only distro be out?

/hwspeedy
